tls-client/main.cpp

```diff
@@ -256,22 +263,32 @@ class HelloHTTPS {
 
 
         /* Read data out of the socket */
-        ret = mbedtls_ssl_read(&_ssl, (unsigned char *) _buffer, sizeof(_buffer));
+        offset = 0;
+        do {
+            ret = mbedtls_ssl_read(&_ssl, (unsigned char *) _buffer + offset, 
+                                   sizeof(_buffer) - offset - 1);
+            if (ret > 0)
+              offset += ret;
+
+            /* Check each of the flags */
+            _buffer[offset] = 0;
+            _got200 = _got200 || strstr(_buffer, HTTPS_OK_STR) != NULL;
+            _gothello = _gothello || strstr(_buffer, HTTPS_HELLO_STR) != NULL;
+        } while (offset < sizeof(_buffer) && (!_got200 || !_gothello) &&
```

Comment:
Here we can omit the `offset < sizeof(_buffer)` condition, because `mbedtls_ssl_read` should check it and throw an error if the data does not fit.